src/shortener/models.py

```diff
@@ -7,3 +8,14 @@ class Url(models.Model):
 
     def __str__(self):
         return '{}:{}'.format(self.url, self.hash)
+
+    # Just for test. Need to be modified.
+    def _generate_hash(self):
+        h = 0
+        for c in self.url:
+            h += ord(c) * ord(random.choice(self.url))
+        return h
+
+    def save(self, *args, **kwargs):
+        self.hash = self._generate_hash()
```

Comment:
Não precisa gerar o hash toda vez que o `.save()` for chamado, dá pra checar se o `self.hash` já existe antes de chamá-lo.